frontend/src/scenes/events/eventsTableLogic.ts

```diff
@@ -33,64 +38,109 @@ const formatEvents = (events, newEvents, apiUrl) => {
     }
     return eventsFormatted
 }
-// props:
-// - fixedFilters
-// - apiUrl = 'api/event/'
-// - live = false
-export const eventsTableLogic = kea({
+
+export interface EventsTableLogicProps {
+    fixedFilters?: FixedFilters
+    apiUrl?: string // = 'api/event/'
+    live?: boolean // = false
+    key?: string
+}
+
+export interface OnFetchEventsSuccess {
+    events: EventType[]
+    hasNext: boolean
+    isNext: boolean
+}
+
+//from visual inspection of lib/api.js
+//we aren't throwing JS Errors
+export interface ApiError {
+    status?: string
+    statusText?: string
+}
+
+export const eventsTableLogic = kea<eventsTableLogicType<ApiError, EventsTableLogicProps, OnFetchEventsSuccess>>({
+    props: {} as EventsTableLogicProps,
     // Set a unique key based on the fixed filters.
     // This way if we move back/forward between /events and /person/ID, the logic is reloaded.
     key: (props) =>
-        (props.fixedFilters ? JSON.stringify(props.fixedFilters) : 'all') +
-        '-' +
-        (props.apiUrl || 'events') +
-        (props.live ? '-live' : '') +
-        props.key,
-
-    actions: () => ({
-        setProperties: (properties) => {
-            if (Array.isArray(properties) && properties.length === 0) {
-                return { properties: {} }
+        [
+            props.fixedFilters ? JSON.stringify(props.fixedFilters) : 'all',
+            props.apiUrl || 'events',
+            props.live ? 'live' : '',
+            props.key,
+        ].join('-'),
+
+    actions: {
+        setProperties: (properties: Record<string, unknown>[]) => {
```

Comment:
There should be a better type for Property filters, something like `properties: AnyPropertyFilter[]` is used at times. Also if the argument here can be an object (and an array), best to also put the object in the type (`x[] | x`)

Reply:
I think I've understood you and resolved this....